File: EDAII x Java/Busqueda/BusquedaLineal.java
public class BusquedaLineal{

	int[] arr = {4,6,9,1,3,2,10,12,1};

	public static void main(String[] args){
		BusquedaLineal bl = new BusquedaLineal();
		int i = bl.busquedaLinealIndice(bl.arr,12);
		System.out.println(i);

		boolean e = bl.busquedaLinealExistente(bl.arr,7);
		System.out.println(e);

		int c = bl.busquedaLinealCantidad(bl.arr,1);
		System.out.println(c);

		i = bl.busquedaLinealRecursivaIndice(bl.arr,3,0,bl.arr.length-1);
		System.out.println(i);

		e = bl.busquedaLinealRecursivaExistente(bl.arr,10,0,bl.arr.length-1);
		System.out.println(e);

		c = bl.busquedaLinealRecursivaCantidad(bl.arr,1,0,bl.arr.length-1);
		System.out.println(c);
	}

	public int busquedaLinealIndice(int[] A, int x){
		int indice=-1;
		for(int i=0; i<A.length; i++){
			if(x==A[i]){
				indice = i;
			}
		}
		return indice;
	}

	public boolean busquedaLinealExistente(int[] A, int x){
		boolean encontrado=false;
		for(int i=0; i<A.length; i++){
			if(x==A[i]){
				encontrado = true;
			}
		}
		return encontrado;
	}

	public int busquedaLinealCantidad(int[] A, int x){
		int cantidad=0;
		for(int i=0; i<A.length; i++){
			if(x==A[i]){
				cantidad ++;
			}
		}
		return cantidad;
	}

	public int busquedaLinealRecursivaIndice(int[] A, int x, int inicio, int fin){
		int indice;
		if(inicio>fin){
			indice = -1;
		} else {
			if(A[inicio]==x){
				indice=inicio;
			} else {
				indice = busquedaLinealRecursivaIndice(A,x,inicio+1,fin);
			}
		}
		return indice;
	}

	public boolean busquedaLinealRecursivaExistente(int[] A, int x, int inicio, int fin){
		boolean encontrado;
		if(inicio>fin){
			encontrado = false;
		} else {
			if(A[inicio]==x){
				encontrado=true;
			} else {
				encontrado = busquedaLinealRecursivaExistente(A,x,inicio+1,fin);
			}
		}
		return encontrado;
	}

	public int busquedaLinealRecursivaCantidad(int[] A, int x, int inicio, int fin){
		int cantidad=0;
		if(inicio>fin){
			cantidad = 0;
		} else {
			if(A[inicio]==x){
				for(int i=inicio; i<=fin; i++){
					if(A[i]==x){
						cantidad ++;
					}
				}
			} else {
				cantidad = busquedaLinealRecursivaCantidad(A,x,inicio+1,fin);
			}
		}
		return cantidad;
	}

}

File: EDAII x Python/Busqueda/BusquedaLineal.py
class BusquedaLineal:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase.'
		self.arr = arr

	def busquedaLinealIndice(self, x):
		'Devuelve la posición en donde se encuentra el elemento a buscar.'
		indice = -1
		for i in range(0, len(self.arr)):
			if x == self.arr[i]:
				indice = i
		return indice

	def busquedaLinealExistente(self, x):
		'Determina si el elemento a buscar se encuentra en el arreglo.'
		encontrado = False
		for i in range(0, len(self.arr)):
			if x == self.arr[i]:
				encontrado = True
		return encontrado

	def busquedaLinealCantidad(self, x):
		'Determina cuantos elementos existen con el valor a buscar.'
		cantidad = 0
		for i in range(0, len(self.arr)):
			if x == self.arr[i]:
				cantidad += 1
		return cantidad

	def busquedaLinealRecursivaIndice(self, x, inicio, fin):
		'Devuelve la posición en donde se encuentra el elemento a buscar.'
		indice = 0
		if inicio > fin:
			indice = -1
		else:
			if self.arr[inicio] == x:
				indice = inicio
			else:
				indice = self.busquedaLinealRecursivaIndice(x,inicio+1,fin)
		return indice

	def busquedaLinealRecursivaExistente(self, x, inicio, fin):
		'Determina si el elemento a buscar se encuentra en el arreglo.'
		encontrado = False
		if inicio > fin:
			encontrado = False
		else:
			if self.arr[inicio] == x:
				encontrado = True
			else:
				encontrado = self.busquedaLinealRecursivaExistente(x,inicio+1,fin)
		return encontrado

	def busquedaLinealRecursivaCantidad(self, x, inicio, fin):
		'Determina cuantos elementos existen con el valor a buscar.'
		cantidad = 0
		if inicio > fin:
			cantidad = 0
		else:
			if self.arr[inicio] == x:
				for i in range(inicio, fin+1):
					if self.arr[i] == x:
						cantidad += 1
			else:
				cantidad = self.busquedaLinealRecursivaCantidad(x,inicio+1,fin)
		return cantidad

# Ejecutando el objeto
programa = BusquedaLineal([4,6,9,1,3,2,10,12,1])
print(programa.busquedaLinealIndice(12))
print(programa.busquedaLinealExistente(7))
print(programa.busquedaLinealCantidad(1))
print(programa.busquedaLinealRecursivaIndice(3,0,len(programa.arr)-1))
print(programa.busquedaLinealRecursivaExistente(10,0,len(programa.arr)-1))
print(programa.busquedaLinealRecursivaCantidad(1,0,len(programa.arr)-1))